src/riscv/lib/src/pvm/linux.rs

```diff
@@ -898,6 +902,50 @@ impl<M: ManagerBase> SupervisorState<M> {
         Ok(0)
     }
 
+    /// Handle `sched_getaffinity` system call. We only support one hart.
+    ///
+    /// See: <https://man7.org/linux/man-pages/man2/sched_getaffinity.2.html>
+    fn handle_sched_getaffinity(
+        &self,
+        core: &mut MachineCoreState<impl MemoryConfig, M>,
+        pid: parameters::NullableProcessId,
+        cpusetsize: parameters::CpuSetSize,
+        mask: parameters::CpuSet,
+    ) -> Result<u64, Error>
+    where
+        M: ManagerReadWrite,
+    {
+        const SINGLE_PROCESS_PID_AFFINITY: u8 = 0b1_u8;
+
+        if pid.0 == 0 {
+            return Ok(SINGLE_PROCESS_PID_AFFINITY as u64);
+        }
+
+        let other_bytes_size = cpusetsize.0.get() - 1;
+
+        if cpusetsize.0.get() > 1 {
+            let mut remaining = other_bytes_size;
+            while remaining >= 8 {
+                remaining -= 8;
+                let address = mask.0 + remaining;
+                core.main_memory.write(address.to_machine_address(), 0u64)?;
+            }
+
+            for i in 0..remaining {
+                let address = mask.0 + i;
+                core.main_memory.write(address.to_machine_address(), 0u8)?;
+            }
+        }
+
+        core.main_memory.write(
+            mask.0.to_machine_address() + other_bytes_size,
```

Comment:
Please compute the virtual address before converting to machine address in order to respect the virtual address arithmetic rules.